src/controller/CHIPDeviceController.cpp

```diff
@@ -1765,51 +1534,41 @@ void DeviceCommissioner::PerformCommissioningStep(DeviceProxy * proxy, Commissio
-        CHIP_ERROR status = CHIP_ERROR_NOT_IMPLEMENTED;
-#endif
-        if (status != CHIP_NO_ERROR)
-        {
-            ChipLogError(Controller, "Unable to find regulatory location, defaulting to outdoor");
-        }
-
+        uint8_t regulatoryLocation                  = EMBER_ZCL_REGULATORY_LOCATION_TYPE_OUTDOOR;
         static constexpr size_t kMaxCountryCodeSize = 3;
         char countryCodeStr[kMaxCountryCodeSize]    = "WW";
         size_t actualCountryCodeSize                = 2;
 
+        bool hasRegulatoryLocation = false;
+        bool hasCountryCode        = false;
 #if CONFIG_DEVICE_LAYER
-        status = DeviceLayer::ConfigurationMgr().GetCountryCode(countryCodeStr, kMaxCountryCodeSize, actualCountryCodeSize);
-#else
-        status = CHIP_ERROR_NOT_IMPLEMENTED;
+        hasRegulatoryLocation = (CHIP_NO_ERROR == DeviceLayer::ConfigurationMgr().GetRegulatoryLocation(regulatoryLocation));
+        hasCountryCode =
+            (CHIP_NO_ERROR ==
+             DeviceLayer::ConfigurationMgr().GetCountryCode(countryCodeStr, kMaxCountryCodeSize, actualCountryCodeSize));
 #endif
-        if (status != CHIP_NO_ERROR)
+
+        if (hasRegulatoryLocation)
+        {
+            ChipLogError(Controller, "Unable to find regulatory location, defaulting to outdoor");
+        }
+
+        if (hasCountryCode)
         {
             ChipLogError(Controller, "Unable to find country code, defaulting to WW");
         }
-        chip::CharSpan countryCode(countryCodeStr, actualCountryCodeSize);
 
-        GeneralCommissioningCluster genCom;
-        genCom.Associate(proxy, 0);
-        genCom.SetRegulatoryConfig(mSuccess.Cancel(), mFailure.Cancel(), regulatoryLocation, countryCode, breadcrumb,
-                                   kCommandTimeoutMs);
+        GeneralCommissioning::Commands::SetRegulatoryConfig::Type request;
+        request.location    = static_cast<EmberAfRegulatoryLocationType>(regulatoryLocation);
+        request.countryCode = CharSpan(countryCodeStr, actualCountryCodeSize);
+        request.breadcrumb  = breadcrumb;
+        request.timeoutMs   = kCommandTimeoutMs;
+        StopSessionIf(SendCommand<GeneralCommissioningCluster>(proxy, request, OnSuccess<decltype(request)::ResponseType>));
     }
     break;
     case CommissioningStage::kDeviceAttestation: {
-        ChipLogProgress(Controller, "Exchanging vendor certificates");
-        // TODO(cecille): Remove the certificates from the CommissioneeDeviceProxy and take from the commissioning parameters.
-        CHIP_ERROR status =
-            SendCertificateChainRequestCommand(reinterpret_cast<CommissioneeDeviceProxy *>(proxy), CertificateType::kPAI);
-        if (status != CHIP_NO_ERROR)
-        {
-            ChipLogError(Controller, "Failed in sending 'Certificate Chain Request' command to the device: err %s",
-                         ErrorStr(status));
-            OnSessionEstablishmentError(status);
-            return;
-        }
+        StopSessionIf(
```

Comment:
And this one should probably be named `StopSessionAndReturnOnFailure` or something...  `StopSessionIf` is really confusing when there is no boolean being passed to it.

Reply:
I have removed `StopSessionIf`.